controller/status.go

```diff
@@ -0,0 +1,81 @@
+package controller
+
+import (
+	"strings"
+
+	"github.com/giantswarm/formica/common"
+	"github.com/giantswarm/formica/fleet"
+)
+
+type UnitStatusList []fleet.UnitStatus
+
+func (usl UnitStatusList) Group() ([]fleet.UnitStatus, error) {
+	matchers := map[string]struct{}{}
+	newList := []fleet.UnitStatus{}
+
+	for _, us := range usl {
+		// Group unit status
+		grouped, suffix, err := groupUnitStatus(usl, us)
+		if err != nil {
+			return nil, maskAny(err)
+		}
+
+		// Prevent doubled aggregation.
+		if _, ok := matchers[suffix]; ok {
+			continue
+		}
+		matchers[suffix] = struct{}{}
+
+		// Aggregate.
+		if allStatesEqual(grouped) {
+			newStatus := grouped[0]
+			newStatus.Name = "*"
+			newList = append(newList, newStatus)
+		} else {
+			newList = append(newList, grouped...)
+		}
+	}
+
+	return newList, nil
+}
+
+func groupUnitStatus(usl []fleet.UnitStatus, groupMember fleet.UnitStatus) ([]fleet.UnitStatus, string, error) {
+	ID, err := common.SliceID(groupMember.Name)
+	if err != nil {
+		return nil, "", maskAny(invalidUnitStatusError)
+	}
+
+	newList := []fleet.UnitStatus{}
+	for _, us := range usl {
+		exp := common.ExtExp.ReplaceAllString(us.Name, "")
+		if !strings.HasSuffix(exp, ID) {
+			continue
+		}
+
+		newList = append(newList, us)
+	}
+
+	return newList, ID, nil
+}
+
+func allStatesEqual(usl []fleet.UnitStatus) bool {
+	for _, us1 := range usl {
+		for _, us2 := range usl {
+			if us1.Current != us2.Current {
+				return false
```

Comment:
looks like we only have test cases for the `return true`  case at the end. Do we want tests for the other cases in here?